apps/rewards/app/components/App/App.js

```diff
@@ -1,27 +1,51 @@
-import { AragonApp, observe, SidePanel } from '@aragon/ui'
+import { AragonApp, observe, SidePanel, TabBar } from '@aragon/ui'
 import PropTypes from 'prop-types'
 import React from 'react'
 import styled from 'styled-components'
 import { Title } from '../Shared'
+import { Empty } from '../Card'
 
 const ASSETS_URL = 'aragon-ui-assets/'
 
 class App extends React.Component {
   static propTypes = {
     app: PropTypes.object.isRequired,
-    accounts: PropTypes.arrayOf(PropTypes.object),
+    rewards: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  state = {
+    selected: 0,
+    rewardsEmpty: this.props.rewards === undefined || this.props.rewards === []
```

Comment:
This way you may need extra functionality to track changes in props - if new rewards are added rewardsEmpty will need to be explicitly updated. Alternatively this.props.rewards === undefined can be checked directly in render() of App or, ultimately, in renders of specific tabs.

Reply:
The new `Overview` and MyRewards` components now contain all empty-case logic, so this is much simpler